### DISCUSSION TITLE
How to delete a binding

### DISCUSSION THREAD

**Discussion by nielsbosma** [AvaloniaUI/Avalonia#7157] [Q&A] [answered]
In WPF we have:

BindingOperations.IsDataBound(this, ItemsCountProperty)
and
BindingOperations.ClearBinding(this, ItemsCountProperty);

How can I achieve the same with Avalonia?

**Comment by kekekeks** [comment, marked as answer]
`AvaloniaObject.Bind` returns `IDisposable`. Disposing it destroys the binding.

**Reply by nielsbosma** [reply]
Thanks!

**Reply by fhaag** [reply]
Can I somehow obtain that `IDisposable` instance later on?

Context: I am receiving an object reference X for a control Y. If X is not `null`, I want to bind several properties of Y to various properties of X. **If X is `null`, I want to remove any existing bindings.** I do not want (and do not see a convenient way) to store the binding references in between, as this is happening in a static helper method.

**Reply by maxkatz6** [reply]
You can also call `AvaloniaObject.ClearValue(Property)`. There is not way to get binding disposable later on.
